### PR TITLE
[llvm] Deprecate Type::getPointerTo()

### PR DESCRIPTION
**Pull request by JOE1994**
Replace remaining uses of `llvm::Type::getPointerTo()` and deprecate it.

It's no longer needed with opaque pointers in LLVM. It may rather confuse new contributors that LLVM has typed pointers.

### CODE REVIEW

File: llvm/include/llvm/IR/Type.h [hunk 1]
```diff
@@ -471,8 +471,9 @@ class Type {
   static Type *getWasm_FuncrefTy(LLVMContext &C);
 
   /// Return a pointer to the current type. This is equivalent to
-  /// PointerType::get(Foo, AddrSpace).
+  /// PointerType::get(Ctx, AddrSpace).
   /// TODO: Remove this after opaque pointer transition is complete.
+  [[deprecated("Use PointerType::get instead")]]
```
Comment (zwuis) [inline llvm/include/llvm/IR/Type.h:476]: Can we use `LLVM_DEPRECATED` in llvm/Support/Compiler.h?

**Comment by JOE1994** [comment]
gentle ping

File: clang/lib/CodeGen/CGObjCMac.cpp
Comment (nikic) [inline clang/lib/CodeGen/CGObjCMac.cpp:1]: Please land the remaining removals separately.

**Review by nikic — APPROVED** [review]
LGTM

**Pull request merged**